### DISCUSSION TITLE
Self signed certificate with OQS from C

### DISCUSSION THREAD

**Discussion by GustavoBodi** [open-quantum-safe/liboqs#1698] [Q&A] [answered]
I would like to know how to create a self signed X509 certificate (without using the CLI), with OpenSSL 3.0,  liboqs and the provider. In my specific case I can't use the provider from the CLI nor use the 1.1.1 branch. In addition, is it possible to call the provider programmatically from C code for dealing with the certificate? Any guidance would be greatly appreciated.

**Comment by baentsch** [comment, marked as answer]
There's nothing special here: Simply use the usual OpenSSL APIs as you would for doing this with classic crypto sig algs. The only thing that changes is that you specify use of [any of the supported PQC sigalgs](https://github.com/open-quantum-safe/oqs-provider?tab=readme-ov-file#signature-algorithms) instead of say "rsa2048" (and of course, ensure [by any of the usual ways](https://github.com/open-quantum-safe/oqs-provider/blob/main/USAGE.md#activation) that `oqsprovider` is loaded -- alongside the `default` provider). Please also note that `oqsprovider` is only used by `openssl` logic, never directly by an app.

**Reply by GustavoBodi** [reply]
Thanks a lot, its working now. The problem was that I misunderstood how OpenSSL loaded and used providers.
